### ISSUE TITLE
Create acquisition functions

### ISSUE THREAD

**Issue by patel-zeel** [sustainability-lab/ASTRA#3]
- [x] Create a folder structure for acquisition functions (@patel-zeel)
- [ ] Add asserts to ensure only supported pairs of acquisitions and strategies work together
- [ ] Max. Entropy - @rishabh-mondal
- [ ] BALD - @aditiagarwal-02
- [ ] Max. variance ratios - @Khush24Shah
- [ ] Mean STD - @VannshJani
- [ ] Learning to Loss - @Madhav-Kanda
- [ ] MC Entropy - @dhruvpatel144
- [ ] MC BALD -  @haikookhandor
- [ ] Diversity (for loop)- @jaiswalsuraj487
- [ ] BatchBALD - @patel-zeel

Create a separate file for each of these strategies in `astra/torch/al/acquisitions` folder. See for example, https://github.com/sustainability-lab/ASTRA/blob/main/astra/torch/al/acquisitions/random.py

Your acquisition function should inherit from all base classes it supports, for example, `Entropy` class will support `MCAcquisition` and `EnsembleAcquisition`.

**Comment by patel-zeel** [comment]
I have added [`DeterministicStrategy`](https://github.com/sustainability-lab/ASTRA/blob/main/astra/torch/al/strategies/deterministic.py) now. Acquisitions such as Entropy should check if the input is 3d `(mc_dim, pool_dim, n_classes)` and take mean over the first dimension. If input is 2d (pool_dim, n_classes), simply compute the acquisition and return the scores.

**Comment by aditiagarwal-02** [comment]
Since MC BALD and BALD have been assigned differently, BALD will only inherit from DeterministicStratergy?

**Comment by patel-zeel** [comment]
@aditiagarwal-02 No, you can inherit from all relevant classes at once. I have updated the assignment accordingly.
